flink-java/src/test/java/org/apache/flink/api/java/ExecutorDiscoveryAndJobClientTest.java

```diff
@@ -68,19 +87,39 @@ private JobExecutionResult executeTestJobBasedOnConfig(final Configuration confi
 	 */
 	public static class IDReportingExecutorFactory implements ExecutorFactory {
 
-		public static final String ID = "test-executor-A";
-
 		@Override
 		public boolean isCompatibleWith(@Nonnull Configuration configuration) {
-			return ID.equals(configuration.get(DeploymentOptions.TARGET));
+			return EXEC_NAME.equals(configuration.get(DeploymentOptions.TARGET));
 		}
 
 		@Override
 		public Executor getExecutor(@Nonnull Configuration configuration) {
 			return (pipeline, executionConfig) -> {
 				final Map<String, OptionalFailure<Object>> res = new HashMap<>();
-				res.put(DeploymentOptions.TARGET.key(), OptionalFailure.of(ID));
-				return new JobExecutionResult(new JobID(), 12L, res);
+				res.put(DeploymentOptions.TARGET.key(), OptionalFailure.of(EXEC_NAME));
+
+				return CompletableFuture.completedFuture(new JobClient(){
```

Comment:
`) {`

Reply:
I think we can introduce a `TestingJobClient` here. A dummy implementation can be used as this, and we survive from multiple effectively-identical anonymous subclasses.